compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVectorLowering.cpp

```diff
@@ -0,0 +1,135 @@
+  LLVMCPUVectorLoweringPass(const LLVMCPUVectorLoweringPassOptions &options) {
+    this->splitVectorTransfersTo = options.splitVectorTransfersTo;
+    this->lowerVectorTransposeTo = options.lowerVectorTransposeTo;
+    this->lowerVectorTransposeToAVX2 = options.lowerVectorTransposeToAVX2;
+    this->lowerVectorMultiReductionTo = options.lowerVectorMultiReductionTo;
+    this->lowerVectorContractionTo = options.lowerVectorContractionTo;
+    this->unrollVectorTransfers = options.unrollVectorTransfers;
+    this->maxTransferRank = options.maxTransferRank;
+  }
+
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<tensor::TensorDialect, linalg::LinalgDialect,
+                    vector::VectorDialect>();
+  }
+  void runOnOperation() override;
+};
+
+void LLVMCPUVectorLoweringPass::runOnOperation() {
+  // Lower high level vector operations like contract or multidim reduce ops
+  // to lower level vector ops.
+  {
+    vector::VectorTransposeLowering vectorTransposeLowering =
+        llvm::StringSwitch<vector::VectorTransposeLowering>(
+            lowerVectorTransposeTo.getValue())
+            .Case("eltwise", vector::VectorTransposeLowering::EltWise)
+            .Case("flat_transpose", vector::VectorTransposeLowering::Flat)
+            .Case("shuffle", vector::VectorTransposeLowering::Shuffle)
+            .Default(vector::VectorTransposeLowering::EltWise);
+    vector::VectorMultiReductionLowering vectorMultiReductionLowering =
+        llvm::StringSwitch<vector::VectorMultiReductionLowering>(
+            lowerVectorMultiReductionTo.getValue())
+            .Case("innerreduction",
+                  vector::VectorMultiReductionLowering::InnerReduction)
+            .Default(vector::VectorMultiReductionLowering::InnerParallel);
+    vector::VectorContractLowering vectorContractLowering =
+        llvm::StringSwitch<vector::VectorContractLowering>(
+            lowerVectorContractionTo.getValue())
+            .Case("matrixintrinsics", vector::VectorContractLowering::Matmul)
+            .Case("dot", vector::VectorContractLowering::Dot)
+            .Case("outerproduct", vector::VectorContractLowering::OuterProduct)
+            .Default(vector::VectorContractLowering::OuterProduct);
+    vector::VectorTransferSplit vectorTransferSplit =
+        llvm::StringSwitch<vector::VectorTransferSplit>(
+            splitVectorTransfersTo.getValue())
+            .Case("none", vector::VectorTransferSplit::None)
+            .Case("linalg-copy", vector::VectorTransferSplit::LinalgCopy)
+            .Case("vector-transfers",
+                  vector::VectorTransferSplit::VectorTransfer)
+            .Default(vector::VectorTransferSplit::None);
+
+    // Per-function lowering pipeline.
+    vector::VectorTransformsOptions vectorTransformOptions =
+        vector::VectorTransformsOptions()
+            .setVectorTransposeLowering(vectorTransposeLowering)
```

Comment:
transpose options shouldn't be necessary here as we are not lowering them in this step.